--- origsrc/rhythmbox-0.12.5/shell/main.c	2009-09-16 16:04:49.000000000 -0500
+++ src/rhythmbox-0.12.5/shell/main.c	2009-11-09 23:02:17.686405800 -0600
@@ -326,13 +326,7 @@ main (int argc, char **argv)
 	} else {
 
 		rb_profile_start ("mainloop");
-#ifdef ENABLE_PYTHON
-		pyg_begin_allow_threads;
-#endif
 		gtk_main ();
-#ifdef ENABLE_PYTHON
-		pyg_end_allow_threads;
-#endif
 		rb_profile_end ("mainloop");
 
 		rb_debug ("out of toplevel loop");
