--- origsrc/rhythmbox-0.10.1/Makefile.am	2007-02-24 17:28:10.000000000 -0600
+++ src/rhythmbox-0.10.1/Makefile.am	2007-07-26 15:53:29.826815400 -0500
@@ -2,11 +2,13 @@
 
 AUTOMAKE_OPTIONS = 1.7
 
+ACLOCAL_AMFLAGS = -I macros
+
 DISTCHECK_CONFIGURE_FLAGS = --disable-schemas-install --enable-gtk-doc --disable-scrollkeeper
 distuninstallcheck_listfiles = find . -type f -print | grep -v '^\./var/scrollkeeper'
 
 SUBDIRS = macros lib metadata rhythmdb widgets sources podcast \
-	backends corelib plugins bindings shell remote data po help tests doc
+	backends bindings corelib plugins shell remote data po help tests doc
 
 EXTRA_DIST = 			\
 	autogen.sh		\
--- origsrc/rhythmbox-0.10.1/backends/gstreamer/Makefile.am	2007-02-24 17:28:09.000000000 -0600
+++ src/rhythmbox-0.10.1/backends/gstreamer/Makefile.am	2007-06-05 14:06:24.272972500 -0500
@@ -8,6 +8,7 @@
 	$(NULL)
 
 librbbackendsgstreamer_la_LIBADD =			\
+	$(GNOME_MEDIA_PROFILES_LIBS)			\
 	$(RHYTHMBOX_LIBS)
 
 librbbackendsgstreamer_la_LDFLAGS = -export-dynamic
--- origsrc/rhythmbox-0.10.1/configure.ac	2007-05-28 08:17:26.000000000 -0500
+++ src/rhythmbox-0.10.1/configure.ac	2007-06-05 12:35:36.010650500 -0500
@@ -780,7 +780,7 @@
 dnl Plugins
 dnl ================================================================
 
-PLUGIN_LIBTOOL_FLAGS="-module -avoid-version"
+PLUGIN_LIBTOOL_FLAGS="-module -avoid-version -no-undefined"
 AC_SUBST(PLUGIN_LIBTOOL_FLAGS)
 
 PLUGINDIR='${libdir}/rhythmbox/plugins'
@@ -834,7 +834,7 @@
 EOF
 
 	if /bin/sh ../libtool --mode=compile ${CC} $PYTHON_CFLAGS -c testpython.c >/dev/null 2>&1 && \
-		/bin/sh ../libtool --mode=link ${CC} -o testpython.la -rpath `pwd` -module -avoid-version $PYTHON_LIB_LOC testpython.lo $PYTHON_LIBS $PYTHON_EXTRA_LIBS >/dev/null 2>&1 && \
+		/bin/sh ../libtool --mode=link ${CC} -o testpython.la -rpath `pwd` -module -avoid-version -no-undefined $PYTHON_LIB_LOC testpython.lo $PYTHON_LIBS $PYTHON_EXTRA_LIBS >/dev/null 2>&1 && \
 		grep 'dlname.*testpython' testpython.la >/dev/null 2>&1; then
 		result=yes
 	else
--- origsrc/rhythmbox-0.10.1/corelib/Makefile.am	2007-02-24 17:27:20.000000000 -0600
+++ src/rhythmbox-0.10.1/corelib/Makefile.am	2007-07-26 15:54:56.048876200 -0500
@@ -16,10 +16,24 @@
 
 librhythmbox_core_la_LIBADD = 				\
 	$(top_builddir)/sources/libsources.la	        \
+	$(top_builddir)/sources/libsourcesimpl.la	\
 	$(top_builddir)/metadata/librbmetadata.la	\
+	$(top_builddir)/podcast/librbpodcast.la		\
 	$(top_builddir)/widgets/librbwidgets.la         \
 	$(top_builddir)/backends/librbbackends.la	\
+	$(top_builddir)/rhythmdb/librhythmdb.la		\
+	$(top_builddir)/plugins/librbplugins.la		\
+	$(top_builddir)/shell/librbshell.la			\
 	$(top_builddir)/lib/librb.la
 
-librhythmbox_core_la_LDFLAGS = -export-dynamic
+if ENABLE_PYTHON
+librhythmbox_core_la_LIBADD += $(top_builddir)/bindings/python/rb.la
+endif
 
+librhythmbox_core_la_LDFLAGS = -export-dynamic -no-undefined
+
+$(top_builddir)/plugins/librbplugins.la:
+	cd $(top_builddir)/plugins; $(MAKE) librbplugins.la
+	 
+$(top_builddir)/shell/librbshell.la:
+	cd $(top_builddir)/shell; $(MAKE) librbshell.la
--- origsrc/rhythmbox-0.10.1/plugins/Makefile.am	2007-02-24 17:27:47.000000000 -0600
+++ src/rhythmbox-0.10.1/plugins/Makefile.am	2007-07-26 15:46:19.744461700 -0500
@@ -89,6 +89,8 @@
 	$(NO_STRICT_ALIASING_CFLAGS)	\
 	$(PYGTK_CFLAGS)			\
 	$(PYTHON_CFLAGS)
+
+librbplugins_la_LIBADD = $(PYTHON_LIBS)
 endif
 
 
--- origsrc/rhythmbox-0.10.1/shell/Makefile.am	2007-02-24 17:27:29.000000000 -0600
+++ src/rhythmbox-0.10.1/shell/Makefile.am	2007-07-26 15:52:22.964804400 -0500
@@ -92,13 +92,9 @@
 	rb-tray-icon.c					\
 	rb-tray-icon.h
 
-rhythmbox_LDADD =					\
-	librbshell.la					\
-	$(top_builddir)/corelib/librhythmbox-core.la	\
-	$(top_builddir)/sources/libsourcesimpl.la	\
-	$(top_builddir)/podcast/librbpodcast.la	        \
-	$(top_builddir)/plugins/librbplugins.la		\
-	$(top_builddir)/rhythmdb/librhythmdb.la
+rb-shell.lo: $(tab_files)
+
+rhythmbox_LDADD = $(top_builddir)/corelib/librhythmbox-core.la
 
 if USE_LIBSOUP
 rhythmbox_LDADD += $(SOUP_LIBS)
@@ -110,10 +106,6 @@
 INCLUDES += $(NOTIFY_CFLAGS)
 endif
 
-if ENABLE_PYTHON
-rhythmbox_LDADD += $(PYTHON_LIBS) $(top_builddir)/bindings/python/rb.la
-endif
-
 if ENABLE_TRACK_TRANSFER
 rhythmbox_LDADD += $(GNOME_MEDIA_PROFILES_LIBS)
 INCLUDES += $(GNOME_MEDIA_PROFILES_CFLAGS)
@@ -148,7 +140,7 @@
 	$(LIBNAUTILUS_BURN_LIBS)			\
 	$(RHYTHMBOX_LIBS)
 
-rhythmbox_LDFLAGS = -export-dynamic -no-undefined
+rhythmbox_LDFLAGS = -export-dynamic
 
 
 BUILT_SOURCES += $(tab_files)
--- origsrc/rhythmbox-0.10.1/sources/Makefile.am	2007-02-24 17:27:24.000000000 -0600
+++ src/rhythmbox-0.10.1/sources/Makefile.am	2007-07-26 14:24:37.144119100 -0500
@@ -60,6 +60,7 @@
 
 libsources_la_LDFLAGS = -export-dynamic
 libsourcesimpl_la_LDFLAGS = -export-dynamic
+libsourcesimpl_la_LIBADD = $(TOTEM_PLPARSER_LIBS)
 
 if USE_LIBSOUP
 INCLUDES += $(SOUP_CFLAGS)
