--- origsrc/rhythmbox-0.11.6/Makefile.am	2008-07-02 04:52:03.000000000 -0500
+++ src/rhythmbox-0.11.6/Makefile.am	2008-12-29 23:29:54.425000000 -0600
@@ -6,7 +6,7 @@
 distuninstallcheck_listfiles = find . -type f -print | grep -v '^\./var/scrollkeeper'
 
 SUBDIRS = macros lib metadata rhythmdb widgets sources podcast \
-	backends corelib plugins bindings shell remote data po help tests doc
+	backends bindings corelib plugins shell remote data po help tests doc
 
 EXTRA_DIST = 			\
 	autogen.sh		\
--- origsrc/rhythmbox-0.11.6/backends/gstreamer/Makefile.am	2008-07-02 03:51:58.000000000 -0500
+++ src/rhythmbox-0.11.6/backends/gstreamer/Makefile.am	2008-12-29 23:09:35.393750000 -0600
@@ -29,6 +29,7 @@
 librbbackendsgstreamer_la_SOURCES +=	\
 	rb-encoder-gst.h		\
 	rb-encoder-gst.c
+librbbackendsgstreamer_la_LIBADD += $(GNOME_MEDIA_PROFILES_LIBS)
 endif
 
 if USE_GSTREAMER_0_10_XFADE
--- origsrc/rhythmbox-0.11.6/configure.ac	2008-07-08 04:08:38.000000000 -0500
+++ src/rhythmbox-0.11.6/configure.ac	2008-12-29 23:31:16.612500000 -0600
@@ -22,6 +22,7 @@
 AC_PROG_CXX
 AC_STDC_HEADERS
 AM_PROG_LIBTOOL()
+m4_ifdef([LT_OUTPUT],[LT_OUTPUT])
 AC_C_BIGENDIAN
 AC_CHECK_SIZEOF(long)
 
@@ -779,7 +780,7 @@
 dnl Plugins
 dnl ================================================================
 
-PLUGIN_LIBTOOL_FLAGS="-module -avoid-version"
+PLUGIN_LIBTOOL_FLAGS="-module -avoid-version -no-undefined"
 AC_SUBST(PLUGIN_LIBTOOL_FLAGS)
 
 PLUGINDIR='${libdir}/rhythmbox/plugins'
@@ -842,7 +843,7 @@
 EOF
 
 	if /bin/sh ../libtool --mode=compile ${CC} $PYTHON_CFLAGS -c testpython.c >/dev/null 2>&1 && \
-		/bin/sh ../libtool --mode=link ${CC} -o testpython.la -rpath `pwd` -module -avoid-version $PYTHON_LIB_LOC testpython.lo $PYTHON_LIBS $PYTHON_EXTRA_LIBS >/dev/null 2>&1 && \
+		/bin/sh ../libtool --mode=link ${CC} -o testpython.la -rpath `pwd` -module -avoid-version -no-undefined $PYTHON_LIB_LOC testpython.lo $PYTHON_LIBS $PYTHON_EXTRA_LIBS >/dev/null 2>&1 && \
 		grep 'dlname.*testpython' testpython.la >/dev/null 2>&1; then
 		result=yes
 	else
--- origsrc/rhythmbox-0.11.6/corelib/Makefile.am	2008-07-02 03:51:58.000000000 -0500
+++ src/rhythmbox-0.11.6/corelib/Makefile.am	2008-12-30 00:51:18.534375000 -0600
@@ -6,12 +6,22 @@
 librhythmbox_core_la_SOURCES = 
 
 librhythmbox_core_la_LIBADD = 				\
+	$(top_builddir)/shell/librbshell.la			\
 	$(top_builddir)/sources/libsources.la	        \
+	$(top_builddir)/sources/libsourcesimpl.la	\
 	$(top_builddir)/metadata/librbmetadata.la	\
 	$(top_builddir)/widgets/librbwidgets.la         \
 	$(top_builddir)/rhythmdb/librhythmdb.la		\
+	$(top_builddir)/podcast/librbpodcast.la		\
+	$(top_builddir)/plugins/librbplugins.la		\
 	$(top_builddir)/backends/librbbackends.la	\
 	$(top_builddir)/lib/librb.la
 
-librhythmbox_core_la_LDFLAGS = -export-dynamic
+if ENABLE_PYTHON
+librhythmbox_core_la_LIBADD += $(top_builddir)/bindings/python/rb.la $(PYTHON_LIBS)
+endif
 
+librhythmbox_core_la_LDFLAGS = -export-dynamic -no-undefined
+
+$(top_builddir)/plugins/librbplugins.la $(top_builddir)/shell/librbshell.la:
+	cd $(@D); $(MAKE) $(@F)
--- origsrc/rhythmbox-0.11.6/plugins/generic-player/Makefile.am	2008-07-02 03:52:00.000000000 -0500
+++ src/rhythmbox-0.11.6/plugins/generic-player/Makefile.am	2008-12-30 01:00:12.128125000 -0600
@@ -17,6 +17,7 @@
 endif
 
 libgeneric_player_la_LDFLAGS = 				\
+	$(top_builddir)/corelib/librhythmbox-core.la	\
 	$(TOTEM_PLPARSER_LIBS)				\
 	$(PLUGIN_LIBTOOL_FLAGS)
 
--- origsrc/rhythmbox-0.11.6/plugins/power-manager/Makefile.am	2008-07-02 03:52:00.000000000 -0500
+++ src/rhythmbox-0.11.6/plugins/power-manager/Makefile.am	2008-12-30 01:07:16.768750000 -0600
@@ -6,6 +6,8 @@
 
 libpower_manager_la_LDFLAGS = $(PLUGIN_LIBTOOL_FLAGS)
 
+libpower_manager_la_LIBADD = $(top_builddir)/corelib/librhythmbox-core.la
+
 INCLUDES = 						\
         -DGNOMELOCALEDIR=\""$(datadir)/locale"\"        \
 	-DG_LOG_DOMAIN=\"Rhythmbox\"		 	\
--- origsrc/rhythmbox-0.11.6/plugins/sample-vala/Makefile.am	2008-07-02 04:05:43.000000000 -0500
+++ src/rhythmbox-0.11.6/plugins/sample-vala/Makefile.am	2008-12-30 01:17:23.862500000 -0600
@@ -11,6 +11,8 @@
 	
 libsample_vala_la_LDFLAGS = $(PLUGIN_LIBTOOL_FLAGS)
 
+libsample_vala_la_LIBADD = $(top_builddir)/corelib/librhythmbox-core.la
+
 libsample_vala_la_CFLAGS = $(WNOERROR_CFLAGS)
 
 INCLUDES = 						\
--- origsrc/rhythmbox-0.11.6/plugins/visualizer/Makefile.am	2008-07-02 03:52:01.000000000 -0500
+++ src/rhythmbox-0.11.6/plugins/visualizer/Makefile.am	2008-12-30 01:11:25.940625000 -0600
@@ -12,6 +12,7 @@
 libvisualizer_la_LDFLAGS = $(PLUGIN_LIBTOOL_FLAGS)
 
 libvisualizer_la_LIBADD = 				\
+	$(top_builddir)/corelib/librhythmbox-core.la	\
 	-lgstinterfaces-0.10				\
 	$(DBUS_LIBS)					\
 	$(RHYTHMBOX_LIBS)
--- origsrc/rhythmbox-0.11.6/podcast/Makefile.am	2008-07-02 03:52:02.000000000 -0500
+++ src/rhythmbox-0.11.6/podcast/Makefile.am	2008-12-30 00:45:13.018750000 -0600
@@ -19,7 +19,7 @@
 	rb-podcast-manager.c				\
 	rb-podcast-manager.h
 librbpodcast_la_LIBADD =				\
-	librbpodcast_parse.la
+	$(TOTEM_PLPARSER_LIBS)
 
 noinst_PROGRAMS = test-podcast-parse
 test_podcast_parse_SOURCES =				\
--- origsrc/rhythmbox-0.11.6/shell/Makefile.am	2008-07-02 03:52:03.000000000 -0500
+++ src/rhythmbox-0.11.6/shell/Makefile.am	2008-12-30 01:57:57.175000000 -0600
@@ -94,11 +94,7 @@
 	rb-missing-plugins.h
 
 rhythmbox_LDADD =					\
-	librbshell.la					\
-	$(top_builddir)/corelib/librhythmbox-core.la	\
-	$(top_builddir)/sources/libsourcesimpl.la	\
-	$(top_builddir)/podcast/librbpodcast.la	        \
-	$(top_builddir)/plugins/librbplugins.la	
+	$(top_builddir)/corelib/librhythmbox-core.la
 
 if USE_LIBSOUP
 rhythmbox_LDADD += $(SOUP_LIBS)
@@ -112,11 +108,9 @@
 
 if ENABLE_PYTHON
 INCLUDES += $(PYGTK_CFLAGS)  $(PYTHON_CFLAGS)
-rhythmbox_LDADD += $(PYTHON_LIBS) $(top_builddir)/bindings/python/rb.la
 endif
 
 if ENABLE_TRACK_TRANSFER
-rhythmbox_LDADD += $(GNOME_MEDIA_PROFILES_LIBS)
 INCLUDES += $(GNOME_MEDIA_PROFILES_CFLAGS)
 endif
 
@@ -125,6 +119,7 @@
 	-lgstpbutils-0.10
 endif
 
+rb-shell.lo: $(tab_files)
 rb-shell-glue.h: rb-shell.xml Makefile
 	$(LIBTOOL) --mode=execute $(DBUS_GLIB_BIN)/dbus-binding-tool --prefix=rb_shell --mode=glib-server --output=$@ $<
 rb-shell-player-glue.h: rb-shell-player.xml Makefile
@@ -141,7 +136,6 @@
 rhythmbox_LDADD += $(DBUS_LIBS)
 
 rhythmbox_LDADD += \
-	$(top_builddir)/backends/librbbackends.la	\
 	$(GDA_LIBS)					\
 	$(TOTEM_PLPARSER_LIBS)				\
 	$(HAL_LIBS)					\
